test/e2e/e2e_test.go

```diff
@@ -1197,6 +1199,176 @@ var _ = Describe("e2e non-vxlan external gateway and update validation", func()
+			framework.Failf("Unable to write CRD config to disk: %v", err)
+		}
+		defer func() {
+			if err := os.Remove(egressFirewallYamlFile); err != nil {
+				framework.Logf("Unable to remove the CRD config from disk: %v", err)
+			}
+		}()
+		// create the CRD config parameters
+		applyArgs := []string{
+			"apply",
+			frameworkNsFlag,
+			"-f",
+			egressFirewallYamlFile,
+		}
+		framework.Logf("Applying EgressFirewall configuration: %s ", applyArgs)
+		// apply the egress firewall configuration
+		framework.RunKubectlOrDie(applyArgs...)
+		// non-ha ci mode runs a set of kind nodes prefixed with ovn-worker
+		ciWorkerNodeSrc := ovnWorkerNode
+		if haMode {
+			// ha ci mode runs a named set of nodes with a prefix of ovn-control-plane
+			ciWorkerNodeSrc = ovnHaWorkerNode
+		}
+		nodeIP, err := runCommand("docker", "inspect", "-f", ciNetworkFlag, ciWorkerNodeSrc)
+		if err != nil {
+			framework.Failf("failed to get the node ip address from node %s %v", ciWorkerNodeSrc, err)
+		}
+		nodeIP = strings.TrimSuffix(nodeIP, "\n")
+		if ip := net.ParseIP(nodeIP); ip == nil {
+			framework.Failf("Unable to retrieve a valid address from container %s with inspect output of %s", ciWorkerNodeSrc, nodeIP)
+		}
+		framework.Logf("the pod side node is %s and the source node ip is %s", ciWorkerNodeSrc, nodeIP)
+		podCIDR, err := getNodePodCIDR(ciWorkerNodeSrc)
+		if err != nil {
+			framework.Failf("Error retrieving the pod cidr from %s %v", ciWorkerNodeSrc, err)
+		}
+		framework.Logf("the pod cidr for node %s is %s", ciWorkerNodeSrc, podCIDR)
+		// create the pod that will be used as the source for the connectivity test
+		createGenericPod(f, srcPodName, ciWorkerNodeSrc, command)
+		// wait for pod setup to return a valid address
+		wait.PollImmediate(retryInterval, retryTimeout, func() (bool, error) {
+			kubectlOut, err := getPodAddress(srcPodName, f.Namespace.Name)
+			validSrcIP = net.ParseIP(kubectlOut)
+			if validSrcIP == nil {
+				return false, nil
+			}
+			if err != nil {
+				return false, nil
+			}
+			return true, nil
+		})
+		framework.Logf("Source pod is %s with an IP of %s ", srcPodName, validSrcIP)
+		// Fail the test if no address is ever retrieved
+		if validSrcIP == nil {
+			framework.Failf("Warning: Failed to get an IP for the source pod %s, test will fail", srcPodName)
+		}
+		// Verify the remote host/port as explicitly allowed by the firewall policy is reachable
+		By(fmt.Sprintf("Verifying connectivity to an explicitly allowed host %s is permitted as defined by the external firewall policy", exFWPermitTcpDnsDest))
+		_, err = framework.RunKubectl("exec", srcPodName, frameworkNsFlag, testContainerFlag, "--", "nc", "-vz", "-w", testTimeout, exFWPermitTcpDnsDest, "53")
+		if err != nil {
+			framework.Failf("Failed to connect to the remote host %s from container %s on node %s: %v", exFWPermitTcpDnsDest, ovnContainer, ovnWorkerNode, err)
+		}
+		// Verify the remote host/port as implicitly denied by the firewall policy is not reachable
```

Comment:
check if you can reuse some of the already code in the framework
https://github.com/kubernetes/kubernetes/blob/1c548c328a32c5b6819195760d1abba115ab4969/test/e2e/network/network_policy.go#L75

Reply:
Great point, I went through those and under the hood are spinning up a pod per each connect attempt with netcat. This is just using a single pod for each test to save time. I do think we can break it out into some functions to tighten it up. Will do it on a future PR and refactor some of the other tests unless you think its a show stopper here. Ty!